### PR TITLE
#11 Drinks pickup mechanic

### LINKED ISSUE #11: Pick up beer/potion mechanic
* Give the player three random beer choices
* Limit the amount of time the player has to make a choice

### PR DESCRIPTION
**Pull request by Zlata-K**
- added a pickup mechanic
- fixed trees (colliders)
- illuminated entrances (lights turn off if tavern is closed/in cooldown)

### CODE REVIEW

File: Assets/Scripts/Drinkables/Drink Pickup/TavernEntrance.cs
Comment (ClaudiaLapalme) [inline Assets/Scripts/Drinkables/Drink Pickup/TavernEntrance.cs]: I would make a variable for the illumination and only call `Find` once instead of twice to boost performance

File: Assets/Scripts/UI/PauseMenu.cs
Comment (ClaudiaLapalme) [inline Assets/Scripts/UI/PauseMenu.cs]: Same as above, a variable would be better than two `Find` calls for performance

File: Assets/Scripts/UI/PauseMenu.cs
Comment (ClaudiaLapalme) [inline Assets/Scripts/UI/PauseMenu.cs]: Same as above

File: Assets/Scripts/Drinkables/Drink Pickup/DrinkChoice.cs
Comment (ClaudiaLapalme) [inline Assets/Scripts/Drinkables/Drink Pickup/DrinkChoice.cs]: Missed this earlier, but same `Find` comment as in the rest of the PR

File: Assets/Scripts/Drinkables/Drink Pickup/DrinkChoice.cs
Comment (ClaudiaLapalme) [inline Assets/Scripts/Drinkables/Drink Pickup/DrinkChoice.cs:102]: Setting the timeScale breaks the pause menu, since the pause manu does `1 - Time.timeScale`, so if we enter the bar, the timeScale goes to 0, but if we press pause, then `1 - 0 = 1` so the timeScale in the pause menu is reversed

**Review by ClaudiaLapalme — CHANGES_REQUESTED** [review]
Pretty good! Left a few comments about things. There is one bug that needs to be fixed

File: Assets/Scripts/Drinkables/Drink Pickup/DrinkChoice.cs
Comment (f-michaut) [reply Assets/Scripts/Drinkables/Drink Pickup/DrinkChoice.cs:102]: Maybe we could disable the pause menu while in a bar ?
Or change the pause menu's operation

Comment (ClaudiaLapalme) [reply Assets/Scripts/Drinkables/Drink Pickup/DrinkChoice.cs:102]: I would introduce a boolean to freeze the movement of all of the characters (ex: `if (inBar) return;`) while in the bar menu to allow the Update's to be called (aka the timer of the bar works fine). This would also allow us to pause everything in the pause menu correctly.

File: Assets/Unity Assets/VillagePack/BigHouse/BigHouse.fbm.meta~HEAD
Comment (f-michaut) [inline Assets/Unity Assets/VillagePack/BigHouse/BigHouse.fbm.meta~HEAD]: Thoses files are garbage (*~HEAD, *~main), can you remove them ?

**Review by f-michaut — DISMISSED** [review]

File: Assets/Unity Assets/VillagePack/BigHouse/BigHouse.fbm.meta~HEAD
Comment (ClaudiaLapalme) [reply Assets/Unity Assets/VillagePack/BigHouse/BigHouse.fbm.meta~HEAD]: I agree with this ^

File: Assets/Scripts/Drinkables/Drink Pickup/DrinkChoice.cs
Comment (f-michaut) [reply Assets/Scripts/Drinkables/Drink Pickup/DrinkChoice.cs:102]: We could do that
Or really change the `1 - Time.timeScale` by `menu.selfActive ? 0 : 1`

Comment (ClaudiaLapalme) [reply Assets/Scripts/Drinkables/Drink Pickup/DrinkChoice.cs:102]: I think that that should work, too

**Review by ClaudiaLapalme — APPROVED** [review]
looks good

**Review by ClaudiaLapalme — CHANGES_REQUESTED** [review]
I'll approve once the merge conflicts are resolved

**Review by ClaudiaLapalme — APPROVED** [review]

**Review by Aleoris — APPROVED** [review]
Tested it and works well. Nice job

**Pull request merged**
